kie-api/src/main/java/org/kie/api/internal/utils/ServiceDiscoveryImpl.java

```diff
@@ -115,10 +113,11 @@ private void processKieService(ClassLoader classLoader, String key, String value
                 if ( value.startsWith( "+" ) ) {
                     childServices.computeIfAbsent( serviceName, k -> new ArrayList<>() )
                             .add( newInstance( classLoader, value.substring( 1 ) ) );
+                    log.debug( "Added child Service " + value );
                 } else {
                     int priority = 0;
                     int separatorPos = value.indexOf( ';' );
-                    if (separatorPos > 0) {
+                    if (separatorPos > 0 && separatorPos < value.length()-1) {
```

Comment:
@mariofusco 
Don't you think this 
`
 int separatorPos = value.indexOf( ';' );
 if (separatorPos > 0 && separatorPos < value.length()-1) {`

is a somehow harder to read (two lines of code plus an  _and_ operator) then

`String[] parts = value.split(";")` ?

Out of curiosity, is there any technical/performance reason to prefer the former to the latter ?

Reply:
@gitgabrio I find the 2 options equally readable, but readability is in the eyes of the reader. It's true that there's a small performance cost in using the split, but this is not my main concern here. In general when I know that there are (at most) only 2 parts I use the indexOf while I use the split when there can be 3 or more. In this specific case this also allow me to throw a NFE also in the case when the user write something like mypkg.MyClass;3;4 because it will try to parse "3;4" as priority. If I use the split it will take '3' as priority (unless I don't add a further check to make sure that there at most 2 parts) that is not what I want here.

Reply:
`mypkg.MyClass;3;4 ` is an invalid syntax so you should throw an exception if the array is more than two elements

And yes, `value.split` is far more readable than indexOf

Reply:
@mariofusco @lucamolteni 
I know discussing so much of three lines of code seems too much.
The reason I insist on that, though, it is because IMO behind the scene this modification imply a radical change in the paradigm/semantic of the kie.conf itself, for which the "split" (o something analogue) would be much more suited.
I'll be glad to explain better this point with a personal conversation.

Beside that, as Luca somehow pointed out, we should identify different Exception condition and manage them appropriately: "mypkg.MyClass;3;4" is a different error then "mypkg.MyClass;3,4" or "mypkg.MyClass;3.4" or ""mypkg.MyClass;a": and I can't see a "nice" way to do that with the "valueOf" approach.